### ISSUE TITLE
Export all pages

### ISSUE THREAD

**Issue by Jerome38** [Laravel-Backpack/CRUD#1710]
# Bug report

### What I did: I've enabled the exports button with : 
`$this->crud->enableExportButtons();`
and tried to export a list of elements 

### What I've already tried to fix it: I tried to disable ajax with    
`$this->crud->ajax_table = false;`
that i've seen on an other github issue but it doesn't work

### Backpack, Laravel, PHP, DB version:
Backpack 3.4

Is there any way to be able to launch a full export on all current page of the grid ?

Thx !
Jerome

**Comment by tabacitu** [comment]
@Jerome38 , unfortunately this is a known issue in the current version. 

Since we use Datatables for the export, it will only export the table that is currently visible. We plan to completely rewrite the export functionality, in PHP, so that the exports will be faster & better. But until then, there’s only one way around it: **select “number of items per page” to “All” before you click the export button**. That will make DataTables export all entries. It’s not the prettiest solution since it requires one extra click (and knowing how it works), I know, but it’s _a solution_.

Cheers!

**Comment by amenk** [comment]
@tabacitu Is there any recent update concerning the backend-based export for all pages?

**Comment by tabacitu** [comment]
No improvement, no. It's a DataTables limitation I'm afraid. We'll only get rid of it once we move away from DataTables entirely (v5.0 the soonest, which is definitely 1+ year away).

**Comment by pcipov** [comment]
So, for export, Backpack is totally unusable for now. It's not good. :-(

**Comment by kkboranbay** [comment]
Check out my new Laravel package for exporting data in Laravel Backpack. It supports exporting all or filtered data, sends exports via email, and avoids memory leaks.
Give it a try: https://github.com/kkboranbay/backpack-export

**Comment by kkboranbay** [comment]
@pcipov 
Check out my new Laravel package for exporting data in Laravel Backpack. It supports exporting all or filtered data, sends exports via email, and avoids memory leaks.
Give it a try: https://github.com/kkboranbay/backpack-export

**Comment by dimer47** [comment]
> @pcipov Découvrez mon nouveau package Laravel pour exporter des données dans Laravel Backpack. Il prend en charge l'exportation de toutes les données ou de données filtrées, envoie les exportations par e-mail et évite les fuites de mémoire. Essayez-le : https://github.com/kkboranbay/backpack-export

Can't wait to try! Finally maybe a solution to export all the pages! 🤞🏻

**Comment by dimer47** [comment]
Following up on this — I ended up building a package that tackles this exact problem:

<img width="919" height="282" alt="Image" src="https://github.com/user-attachments/assets/92f43ca2-1c9c-4d20-85d0-5d1577040326" />

**[dimer47/backpack-server-export](https://github.com/dimer47/backpack-server-export)**
It runs the export entirely server-side, so there's no page/row limit from DataTables. It also captures the current filters, search term, sort order, and column visibility from the list view, so the export matches exactly what the user sees (minus the pagination).

Integration is one line in your CRUD controller:

```php
use ServerExportOperation;
```

A "Full Export" dropdown appears in the DataTables toolbar with XLSX, CSV, and Markdown options. For large datasets, it automatically switches to a queue job.

I tested it with @kkboranbay's package too at the time — it takes a different approach (email delivery). Mine is more focused on keeping the flow inside the browser: small exports download directly, large ones go through a background job.

If anyone here is still looking for a solution to this, feel free to try it and let me know how it goes.
